src/main/java/gregtech/common/pipelike/cable/net/EnergyNetWalker.java

```diff
@@ -25,7 +24,7 @@
     }
 
     private final List<RoutePath> routes;
-    private Set<TileEntityCable> pipes = new HashSet<>();
+    private List<TileEntityCable> pipes = new ArrayList<>();
```

Comment:
If this is only used for copying, a `TileEntityCable[]` and copying via `System.arrayCopy` with it would be even better.

Reply:
No, during walking things are added. Or would it be better to use `ArrayUtils.add()` with arrays?

Reply:
yes, `ArrayUtils.add()` is a QoL version of `System.arrayCopy`, same as what `List` does in the backend.